modules/dhcp_common/record.rb

```diff
@@ -45,7 +45,7 @@ def eql?(other)
     end
 
     def ==(other)
-      !other.nil? && self.class == other.class && ip == other.ip && mac == other.mac && subnet == other.subnet && deleteable? == other.deleteable? && options == other.options
+      !other.nil? && self.class == other.class && ip == other.ip && mac == other.mac && subnet == other.subnet && deleteable? == other.deleteable? && options.tap { |option| option.delete(:deleteable) } == other.options.tap { |option| option.delete(:deleteable) }
```

Comment:
This mutates both records, which it shouldn't (options hash will not contain :deleteable key if it was there before the comparison). Any reason you didn't use ```options[:deleteable] == other.options[:deleteable]``` instead?

This raises another question -- what options should be taken into account in equality comparison? For example, filename, nextserver, hostname, and a bunch of vendor-specfic options are used during record creation and were used for equality comparisons before this change.

Reply:
Considering that we assume that reservations are always deleteable, and if we make them so by default (pls. see my comment below), I don't think this change is needed at all.